plugins/woocommerce-blocks/tests/e2e/tests/product-button/product-button.classic_theme.spec.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { expect, test as base } from '@woocommerce/e2e-playwright-utils';
+
+/**
+ * Internal dependencies
+ */
+import { blockData } from './utils';
+import ProductCollectionPage from '../product-collection/product-collection.page';
+
+const test = base.extend< { productCollectionPage: ProductCollectionPage } >( {
+	productCollectionPage: async (
+		{ page, admin, editor, templateApiUtils, editorUtils },
+		use
+	) => {
+		const pageObject = new ProductCollectionPage( {
+			page,
+			admin,
+			editor,
+			templateApiUtils,
+			editorUtils,
+		} );
+		await use( pageObject );
+	},
+} );
+test.describe( `${ blockData.name } Block`, () => {
+	test.beforeEach( async ( { productCollectionPage } ) => {
+		await productCollectionPage.createNewPostAndInsertBlock();
```

Comment:
In the future, do you think we could use a fixture instead of inserting the block in the `beforeEach()`? We had a discussion about this with @sunyatasattva in https://github.com/woocommerce/woocommerce/pull/42903#issuecomment-1863612267 and he convinced me. :slightly_smiling_face:

Reply:
Brilliant idea! Yeah, it makes sense! I switched approached!